kit/inputs/src/UIMaterialTextView/useAutogrow.ts

```diff
@@ -1,88 +1,45 @@
 import * as React from 'react';
-import {
-    TextInput,
-    NativeSyntheticEvent,
-    TextInputChangeEventData,
-    StyleSheet,
-    Platform,
-} from 'react-native';
-
-import { OnHeightChange, useAutogrowTextView } from '../useAutogrowTextView';
-import type { UIMaterialTextViewProps } from './types';
+import type { TextInput } from 'react-native';
+import type { OnHeightChange } from '../useAutogrowTextView';
+import type { UIMaterialTextViewProps, AutogrowAttributes } from './types';
 
 export function useAutogrow(
-    ref: React.Ref<TextInput>,
+    _ref: React.Ref<TextInput>,
     onContentSizeChangeProp: UIMaterialTextViewProps['onContentSizeChange'],
     onChangeProp: UIMaterialTextViewProps['onChange'],
-    multiline: UIMaterialTextViewProps['multiline'],
+    _multiline: UIMaterialTextViewProps['multiline'],
     numberOfLinesProp: UIMaterialTextViewProps['numberOfLines'],
-    onHeightChange?: OnHeightChange,
-) {
-    const {
-        onContentSizeChange: onAutogrowContentSizeChange,
-        onChange: onAutogrowChange,
-        inputHeight,
-        numberOfLines,
-        resetInputHeight,
-    } = useAutogrowTextView(ref, onHeightChange, multiline ? numberOfLinesProp : 1);
+    onHeightChange: OnHeightChange | undefined,
+    _isHovered: boolean,
+    _isFocused: boolean,
+): AutogrowAttributes {
+    const inputHeight = React.useRef(0);
 
     const onContentSizeChange = React.useCallback(
         (event: any) => {
-            if (onAutogrowContentSizeChange) {
-                onAutogrowContentSizeChange(event);
+            if (onHeightChange) {
+                const height = event?.nativeEvent?.contentSize?.height;
+                if (height !== inputHeight.current) {
+                    inputHeight.current = height;
+                    if (typeof height === 'number') {
```

Comment:
Could you please put a comment why you had to check for type of a number, as it's unclear what was the problem

Reply:
I put down the type of the `event` and realized that the check was unnecessary. Removed